source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2736 @@
+      lsid: { $$sessionLsid: *session0 }
+
+
+Test Runner Implementation
+--------------------------
+
+The sections below describe instructions for instantiating the test runner,
+loading each test file, and executing each test within a test file. Test runners
+SHOULD NOT share state created by processing a test file with the processing of
+subsequent test files, and likewise for tests within a test file.
+
+
+Initializing the Test Runner
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The test runner MUST be configurable with a connection string (or equivalent
+configuration), which will be used to initialize the internal MongoClient and
+any `client entities <entity_client_>`_ (in combination with other URI options).
+This specification is not prescriptive about how this information is provided.
+For example, it may be read from an environment variable or configuration file.
+
+Create a new MongoClient, which will be used for internal operations (e.g.
+processing `initialData`_ and `test.outcome <test_outcome_>`_). This is referred
+to elsewhere in the specification as the internal MongoClient.
+
+Determine the server version and topology type using the internal MongoClient.
+This information will be used to evaluate any future `runOnRequirement`_ checks.
+
+The test runner SHOULD terminate any open transactions (see:
+`Terminating Open Transactions`_) using the internal MongoClient before
+executing any tests.
+
+
+Executing a Test File
+~~~~~~~~~~~~~~~~~~~~~
+
+The instructions in this section apply for each test file loaded by the test
+runner.
+
+Test files, which may be YAML or JSON files, MUST be interpreted using an
+`Extended JSON`_ parser. The parser MUST accept relaxed and canonical Extended
+JSON (per `Extended JSON: Parsers <../extended-json.rst#parsers>`__), as test
+files may use either.
+
+Upon loading a file, the test runner MUST read the `schemaVersion`_ field and
+determine if the test file can be processed further. Test runners MAY support
+multiple versions and MUST NOT process incompatible files (as discussed in
+`Schema Version`_).
+
+If `runOnRequirements`_ is specified, the test runner MUST skip the test file
+unless one or more `runOnRequirement`_ objects are satisfied.
+
+For each element in `tests`_, follow the process in `Executing a Test`_.
+
+
+Executing a Test
+~~~~~~~~~~~~~~~~
+
+The instructions in this section apply for each `test`_ occuring in a test file
+loaded by the test runner. After processing a test, test runners SHOULD reset
+any internal state that resulted from doing so. For example, the `Entity Map`_
+created for one test SHOULD NOT be shared with another.
+
+If at any point while executing this test an unexpected error is encountered or
+an assertion fails, the test runner MUST consider this test to have failed and
+SHOULD continue with the instructions in this section to ensure that the test
+environment is cleaned up (e.g. disable fail points, kill sessions) while also
+forgoing any additional assertions.
```

Comment:
foregoing?

Reply:
This is correct as written, as the verb is "forgo". See: here.https://www.merriam-webster.com/words-at-play/usage-of-forego-vs-forgo

Reply:
I learned something today, thank you!